internal/peer/node/throttle_grpc.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package node
+
+import (
+	"context"
+
+	"github.com/hyperledger/fabric/common/semaphore"
+	"github.com/hyperledger/fabric/core/peer"
+	"google.golang.org/grpc"
+)
+
+var grpcSemaphores = make(map[string]Semaphore)
+
+// Semaphore defines the interface that acquires and releases a permit
+type Semaphore interface {
+	// Acquire acquires a permit
+	Acquire(ctx context.Context) error
+	// Release releases a permit.
+	Release()
+}
```

Comment:
Why do we need an interface? You're constructing a concrete type in the unexported constructor and have no tests that replace it with a mock.

Reply:
Forgot to remove - it was there from an old (different) approach. Removed now.